crates/forge/bin/cmd/bind_json.rs

```diff
@@ -0,0 +1,500 @@
+use super::eip712::Resolver;
+use clap::Parser;
+use eyre::{Ok, Result};
+use foundry_cli::{opts::CoreBuildArgs, utils::LoadConfig};
+use foundry_common::{compile::with_compilation_reporter, fs};
+use foundry_compilers::{
+    artifacts::{
+        output_selection::OutputSelection, ContractDefinitionPart, Source, SourceUnit,
+        SourceUnitPart, Sources,
+    },
+    multi::{MultiCompilerLanguage, MultiCompilerParsedSource},
+    project::ProjectCompiler,
+    solc::SolcLanguage,
+    CompilerSettings, Graph, Project,
+};
+use itertools::Itertools;
+use rayon::prelude::*;
+use solang_parser::pt as solang_ast;
+use std::{
+    collections::{BTreeMap, BTreeSet},
+    path::PathBuf,
+    sync::Arc,
+};
+
+foundry_config::impl_figment_convert!(BindJsonArgs, opts);
+
+/// CLI arguments for `forge bind-json`.
+#[derive(Clone, Debug, Parser)]
+pub struct BindJsonArgs {
+    #[command(flatten)]
+    opts: CoreBuildArgs,
+}
```

Comment:
this this also include structs from dependencies?
I think we can make this optional

maybe even some glob filter, but this we can add in a followup